### PR TITLE
[autogluon][build] Autogluon 0.3.0

### PR DESCRIPTION
**Pull request by SergTogul**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: autogluon/training/docker/0.3/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -101,7 +101,7 @@ RUN pip install --no-cache-dir -U \
 && pip uninstall -y jupyter-client jupyter-core jupyter-packaging jupyter-server jupyter-server-proxy \
     jupyterlab jupyterlab-nvdashboard jupyterlab-pygments jupyterlab-server jupyterlab-widgets ipywidgets \
     widgetsnbextension traitlets pydeck panel nbformat nbconvert nbclient pyppeteer matplotlib-inline \
-    ipython-genutils ipython ipykernel torchvision bokeh plotly notebook datashader
+    ipython-genutils ipython ipykernel bokeh notebook datashader
```
Comment (saimidu) [inline autogluon/training/docker/0.3/py3/Dockerfile.cpu:104]: Please break these package names into multiple lines, with one package name per line. It is currently hard to understand what packages are being installed/uninstalled, and whether any package names are repeated.

Also, why is `bokeh` being uninstalled?

Why does this image pass the https://github.com/aws/deep-learning-containers/blob/master/test/dlc_tests/sanity/test_utility_installation.py#L62-L65 test if `bokeh` is not installed in the image?

File: autogluon/training/docker/0.3/py3/cu102/Dockerfile.gpu [hunk 2]
```diff
@@ -50,7 +50,7 @@ RUN source activate rapids \
  && pip uninstall -y jupyter-client jupyter-core jupyter-packaging jupyter-server jupyter-server-proxy \
     jupyterlab jupyterlab-nvdashboard jupyterlab-pygments jupyterlab-server jupyterlab-widgets ipywidgets \
     widgetsnbextension traitlets pydeck panel nbformat nbconvert nbclient pyppeteer matplotlib-inline websockets \
-    ipython-genutils ipython ipykernel torchvision bokeh plotly notebook datashader
+    ipython-genutils ipython ipykernel bokeh notebook datashader
```
Comment (saimidu) [inline autogluon/training/docker/0.3/py3/cu102/Dockerfile.gpu:53]: Same comments as the CPU image.

**Comment by saimidu** [comment]
@SergTogul Why are AutoGluon CI tests not being triggered as a part of this PR?

File: autogluon/training/docker/0.3/py3/Dockerfile.cpu (hunk 1 above)
Comment (SergTogul) [reply autogluon/training/docker/0.3/py3/Dockerfile.cpu:104]: All those packages are uninstalled to make an image as small as possible. 
Re 1 package at 1 string - I understand readability is not prefect. But do you really prefer 40 strings instead of 5?
What's wrong with test_utility_packages_using_import? It checks only SM version.

**Review by saimidu — APPROVED** [review]
